classes/06class/exercises/c06-serverless01/alanlima/src/main.py

```diff
@@ -0,0 +1,91 @@
+import os
+import uuid
+import boto3
+import base64
+import json
+import datetime
+from logging import error
+
+def getParameter(name):
+    ssm = boto3.client('ssm')
+    parameter = ssm.get_parameter(Name=name)
+    print(f"get parameter {parameter}")
+    return parameter['Parameter']['Value']
```

Comment:
A potential problem when fetching parameters directly from SSM in lambdas is throttling.

> Parameter Store launched support for higher throughput last year. You can now make up to 1,000 requests per second if you enable higher throughput setting in your account.
For more information: https://aws.amazon.com/about-aws/whats-new/2019/04/aws_systems_manager_now_supports_use_of_parameter_store_at_higher_api_throughput/

If the value is not sensitive you can store it as an env var (DB_NAME could be there), if it is sensitive you may keep it in memory using a singleton pattern so it will live while the container is alive and will fetch the param again in a new warm start.

Another option would be writing a file to `/tmp` lambda temporary storage.

Reply:
Nice call, I didn't know about that. I did some adjustments in the code and the parameter is now being created/deleted via script.